openmc/plots.py

```diff
@@ -300,27 +452,70 @@ def meshlines(self, meshlines):
         string = 'Plot\n'
-        string += '{0: <16}{1}{2}\n'.format('\tID', '=\t', self._id)
-        string += '{0: <16}{1}{2}\n'.format('\tName', '=\t', self._name)
-        string += '{0: <16}{1}{2}\n'.format('\tFilename', '=\t', self._filename)
-        string += '{0: <16}{1}{2}\n'.format('\tType', '=\t', self._type)
-        string += '{0: <16}{1}{2}\n'.format('\tBasis', '=\t', self._basis)
-        string += '{0: <16}{1}{2}\n'.format('\tWidth', '=\t', self._width)
-        string += '{0: <16}{1}{2}\n'.format('\tOrigin', '=\t', self._origin)
-        string += '{0: <16}{1}{2}\n'.format('\tPixels', '=\t', self._origin)
-        string += '{0: <16}{1}{2}\n'.format('\tColor', '=\t', self._color)
-        string += '{0: <16}{1}{2}\n'.format('\tBackground', '=\t',
-                                            self._background)
-        string += '{0: <16}{1}{2}\n'.format('\tMask components', '=\t',
+        string += '{: <16}=\t{}\n'.format('\tID', self._id)
+        string += '{: <16}=\t{}\n'.format('\tName', self._name)
+        string += '{: <16}=\t{}\n'.format('\tFilename', self._filename)
+        string += '{: <16}=\t{}\n'.format('\tType', self._type)
+        string += '{: <16}=\t{}\n'.format('\tBasis', self._basis)
+        string += '{: <16}=\t{}\n'.format('\tWidth', self._width)
+        string += '{: <16}=\t{}\n'.format('\tOrigin', self._origin)
+        string += '{: <16}=\t{}\n'.format('\tPixels', self._origin)
+        string += '{: <16}=\t{}\n'.format('\tColor by', self._color)
+        string += '{: <16}=\t{}\n'.format('\tBackground', self._background)
+        string += '{: <16}=\t{}\n'.format('\tMask components',
                                             self._mask_components)
-        string += '{0: <16}{1}{2}\n'.format('\tMask background', '=\t',
+        string += '{: <16}=\t{}\n'.format('\tMask background',
                                             self._mask_background)
-        string += '{0: <16}{1}{2}\n'.format('\tCol Spec', '=\t', self._col_spec)
-        string += '{0: <16}{1}{2}\n'.format('\tLevel', '=\t', self._level)
-        string += '{0: <16}{1}{2}\n'.format('\tMeshlines', '=\t',
-                                            self._meshlines)
+        string += '{: <16}=\t{}\n'.format('\tColors', self._colors)
+        string += '{: <16}=\t{}\n'.format('\tLevel', self._level)
+        string += '{: <16}=\t{}\n'.format('\tMeshlines', self._meshlines)
         return string
 
+    @classmethod
+    def from_geometry(cls, geometry, basis='xy', slice_coord=0.):
+        """Return plot that encompasses a geometry.
+
+        Parameters
+        ----------
+        geometry : openmc.Geometry
+            The geometry the base the plot off of
+        basis : {'xy', 'xz', 'yz'}
+            The basis directions for the plot
+        slice_coord : float
+            The level at which the slice plot should be plotted. For example, if
+            the basis is 'xy', this would indicate at what z value is used in
```

Comment:
This reads a little weird, perhaps rephrase as "at what z value" -> "the z value used in the origin"